Feat/graficos falhas marcacoes

gui-benedito opened the pull request:
# | Desenvolver o gráfico de faltas | US SKF-125 #137 | #
# | Desenvolver o gráfico de solicitações de ajuste de ponto | US SKF-126 #138 | #

### O que foi feito: ###
- Criado componentes de gráficos

### Passos para testar: ###
1.  Executar a aplicação que está na branch feat/crud_atrasos
2. Acessar página de Falhas em marcações

### Checklist 
- [ ] Atualizado com a develop
- [ ] Dentro dos critérios de aceitação
- [ ] Adicionado novas dependências
- [ ] Código limpo e comentado
- [ ] Dentro dos padrões de Projeto

ajperes committed:
refactor: Base 'frontend' separada

ajperes committed:
refactor: Arquivos fonte movidos para 'src'

karengoncalves8 committed:
Feat/css config (#2)

* feat: adding colors.css file

* feat: added all the colors

ericloumendes committed:
feat: Adicionando tela de login

ericloumendes committed:
hotfix: Ajustando gerais

- Ajuste das validações do formulário

- Adiçãoda funcionalidade de exibir/esconder senha

- Ajuste das sombras do card e botão

- Ajuste da fonte da página e logo

- tradução dos comentários de código para o português

ericloumendes committed:
hotfix: Ajustes gerais

- Ajustando sombra do card

- Validação dos campos de login

ericloumendes committed:
Merge pull request #16 from SkyFlyTeam/feat/telaLogin

Feat/tela login

BrennoLyrio committed:
feat: criado componente de ponto e sidebar

BrennoLyrio committed:
hotfix: alterando os imports necessários

BrennoLyrio committed:
hotfix: alterações na sidebar e components

-Foi mudado os Icons na versão mobile de fechar e abrir a sidebar -Mudado as páginas de acessode cada cargo -Mudei os components usados envolvendo a sidebar para dentro de custom e ajeitado a pages

andresalerno committed:
feat: criando telas do colaborador

gustavomuraoka committed:
Merge branch 'develop' into feat/telas-colaborador

BrennoLyrio committed:
feat: Tela de Histórico de Pontos

BrennoLyrio committed:
hotfix: lógica melhorada e paginação

gustavomuraoka committed:
feat: página de gerenciar informações sobre empresa e setor

BrennoLyrio committed:
feat: Foi adicionado o envio de email de primeiro acesso

gui-benedito committed:
feat: Criar card de carga diária (#17)

* feat: Criar card de carga diária

* feat: Criar responsividade

* fix: tartando barra excedida

* fix: card carga horaria
ajustado o box-shadow
hora extra acima de jornada de trabalho no mobile, e se 0h não aparece

* fix: Requisitar horas, jornadas e ponto do usuário

* fix: ajustado envio do formulário para o backend

---------

Co-authored-by: karengoncalves8 <karengoncalves128@gmail.com>

karengoncalves8 committed:
style: ajustado detalhes no sidebar css

ericloumendes committed:
Merge branch 'develop' of https://github.com/SkyFlyTeam/BeeOnTime-frontend into feat/navbar

andresalerno committed:
hotfix: adicionando máscara para o CPF

andresalerno committed:
ajustando o campo Cargo do Modal Cadastrar Colaborador

andresalerno committed:
hotfix: ajustando o nível de acesso com o option value

andresalerno committed:
hotfix: ajustando o botão 'Próximo' para a adição do hover

andresalerno committed:
hotfix: ajustando a cor e o hover do botão 'Salvar' - segundo modal

karengoncalves8 committed:
feat: adicionado geração de senha

SarahBatagioti committed:
refactor: substituída geração de senha com biblioteca por lógica customizada

ajperes committed:
refactor: Integrado feat/rotas em 'develop'

ajperes committed:
refactor: Removido pontos de 'debug'

andresalerno committed:
hotfix: tirando a coluna de ações

karengoncalves8 committed:
fix: ajustado toast

karengoncalves8 committed:
Merge branch 'feat/telas-colaborador' of https://github.com/SkyFlyTeam/BeeOnTime-frontend into feat/telas-colaborador

gustavomuraoka committed:
hotfix: arrumando components para maior fidelidade ao modelo proposto no figma

BrennoLyrio committed:
feat: Foi adicionado o PontoService e os diferentes acessos, além de ter corrigido a tabela

ajperes committed:
refactor: getRoleID() e reajuste de metodos.

karengoncalves8 committed:
feat: integrado com o backend

ericloumendes committed:
feat: Proteção de rotas

Feat/rotas

andresalerno committed:
hotfix:ajustes baseado nas sugestões do PR

ericloumendes committed:
hotfix: Integrando card Bater ponto com o backend

ericloumendes committed:
hotfix: Integrando card Bater ponto com o backend

gustavomuraoka committed:
Merge pull request #23 from SkyFlyTeam/feat/gerenciar-empresa

Tela Gerenciar empresa | RNF03 | US-9

gustavomuraoka committed:
hotfix: recaracterização da tela Empresa

gustavomuraoka committed:
Merge pull request #28 from SkyFlyTeam/hotfix/recaracterizacao-empresa

hotfix: recaracterização da tela Empresa

ericloumendes committed:
hotfix: Removendo alertas de erros de status de login

- Adição de um toast no lugar

ericloumendes committed:
Merge branch 'develop' into feat/tela_historico_ponto

ericloumendes committed:
Merge branch 'develop' into feat/email

ericloumendes committed:
hofix: Ajustando rotas para puxar id de usuário conectado

ericloumendes committed:
Merge pull request #29 from SkyFlyTeam/hotfix/Ajusterotas

Hotfix/ajusteRotas

gustavomuraoka committed:
hotfix: acerto de estrutura seguindo a develop

gustavomuraoka committed:
Merge branch 'develop' into hotfix/modal-cadastro-empresa

gustavomuraoka committed:
Merge pull request #30 from SkyFlyTeam/hotfix/modal-cadastro-empresa

hotfix: acerto de estrutura seguindo a develop

karengoncalves8 committed:
fix: ajustado validação do formulário e responsividade

karengoncalves8 committed:
Merge branch 'develop' into feat/telas-colaborador

karengoncalves8 committed:
refactor: excluido pasta inutilizada

ericloumendes committed:
hotfix: Ajustando card ponto para obter o id de usuário conectado

gustavomuraoka committed:
Merge branch 'develop' into hotfix/CardPontoId

gustavomuraoka committed:
Merge pull request #31 from SkyFlyTeam/hotfix/CardPontoId

hotfix: Ajustando card ponto para obter o id de usuário conectado

gustavomuraoka committed:
Merge branch 'develop' into feat/tela_historico_ponto

ericloumendes committed:
hotfix: Ajustes gerais

- Ajustando campos da navbar baseado no nível de acesso do usuário

- Ajustando página inicial

- Ajustando página de edição de empresa

ericloumendes committed:
hotfix: Ajustes gerais

hotfix: Ajustes gerais

ericloumendes committed:
Merge pull request #20 from SkyFlyTeam/feat/tela_historico_ponto

Tela de Histórico de Pontos

ericloumendes committed:
hotfix: Ajuste Horario-ponto

Removendo espediente noturno da tabela

Ajustando rotas da navbar

Ajustando permição de acesso

ericloumendes committed:
hotfix: ajustando tabela responsiva de historico-pontos

ericloumendes committed:
Merge pull request #33 from SkyFlyTeam/hotfix/AjusteHorarioPonto

Hotfix/ajuste horario ponto

karengoncalves8 committed:
fix: ajustado a visualização quando não há dados

ericloumendes committed:
Merge branch 'develop' into feat/telas-colaborador

ericloumendes committed:
Merge pull request #19 from SkyFlyTeam/feat/telas-colaborador

# | Criação telas colaborador/SKF-52 e 79 | (RFN01) | #7

gui-benedito committed:
hotfix: Ajustar modal

ericloumendes committed:
hotfix: Ajustando integrações colaboradores

ericloumendes committed:
Merge pull request #35 from SkyFlyTeam/hotfix/IntegracaoColaboradores

hotfix: Ajustando integrações colaboradores

ericloumendes committed:
Merge branch 'develop' into hotfix/solicitacao

ericloumendes committed:
Merge pull request #34 from SkyFlyTeam/hotfix/solicitacao

hotfix: Ajustar modal

ericloumendes committed:
Merge branch 'develop' into feat/email

ericloumendes committed:
Merge pull request #24 from SkyFlyTeam/feat/email

feat: Foi adicionado o envio de email de primeiro acesso

ericloumendes committed:
fix: implementado o card de horario

ericloumendes committed:
Merge pull request #38 from SkyFlyTeam/hotfix/ajuste-card-horario

fix: implementado o card de horario

gui-benedito committed:
hotfix: Aprovar e recusar
Ajuste na visualização das solicitações de acordo com o token

gui-benedito committed:
hotfix: Comportamento dos botões

gui-benedito committed:
hotfix: Comportamento de renderização

gui-benedito committed:
Merge branch 'develop' into hotfix/solicitacao

gui-benedito committed:
Merge pull request #39 from SkyFlyTeam/hotfix/solicitacao

hotfix: Aprovar e recusar

andresalerno committed:
hotfix: adicionando a coluna de ações

gustavomuraoka committed:
hotfix: reformatando email de primeiro acesso

gustavomuraoka committed:
Merge pull request #41 from SkyFlyTeam/hotfix/reformatação-email

hotfix: reformatando email de primeiro acesso

gustavomuraoka committed:
hotfix: ajustando cores das fontes

gustavomuraoka committed:
Merge pull request #42 from SkyFlyTeam/hotfix/fonte-cores

hotfix: ajustando cores das fontes

gui-benedito committed:
fix: Modal para envio de solicitação

ericloumendes committed:
hotfix: adicionando tela notFound

ericloumendes committed:
Merge pull request #44 from SkyFlyTeam/hotfix/PaginaNotFound

hotfix: adicionando tela notFound

ericloumendes committed:
Merge pull request #43 from SkyFlyTeam/hotfix/enviar-solicitacao

fix: Modal para envio de solicitação

ericloumendes committed:
Merge pull request #40 from SkyFlyTeam/hotfix/ajustes-coluna-acoes

Hotfix: adicionando a coluna de ações

gustavomuraoka committed:
hotfix: reestruturação do email de primeiro acesso para funcionário

ericloumendes committed:
hotfix: Ajustando historico-ponto

gui-benedito committed:
hotfix: Enviar solicitacao

ericloumendes committed:
hotfix: Ajustando edição usuário primordial

ericloumendes committed:
Merge pull request #46 from SkyFlyTeam/hotfix/email-funcionario

hotfix: reestruturação do email de primeiro acesso para funcionário

ericloumendes committed:
Merge pull request #48 from SkyFlyTeam/hotfix/CardEditarFuncionario

Hotfix/card editar funcionario

ericloumendes committed:
Merge pull request #47 from SkyFlyTeam/hotfix/enviar-solicitacao

hotfix: Enviar solicitacao

gui-benedito committed:
hotfix: filtrar solicitação
filtragem de solicitação por usuário e setor, para gestor visualizar apenas seus funcionários

gui-benedito committed:
fix: Modal de solicitações
adicionado toast ao criar, aprovar ou recusar solicitação
ajuste na font do modal
ajuste na largura do input de justivifativa

karengoncalves8 committed:
fix: ajustado o erro na pagina de hist-pontos

karengoncalves8 committed:
Hotfix/jornada flexivel (#51)

* style: ajustado estilo dos cards de inicio funcionario

* fix: ajustado o card de jornada

* hotfix: carga horária

* fix: ajuste do card de ponto

* fix: ajustado botão de ponto e estilo da navbar

---------

Co-authored-by: Eric Lourenco <ericloumendes@gmail.com>
Co-authored-by: Maria Luíza <108702021+Mariauiza@users.noreply.github.com>

karengoncalves8 committed:
fix: ajustado toda a estrutura do projeto (#52)

Co-authored-by: Maria Luíza <108702021+Mariauiza@users.noreply.github.com>

gui-benedito committed:
Merge branch 'develop' into hotfix/filtrar-solicitacao

gui-benedito committed:
Merge pull request #53 from SkyFlyTeam/hotfix/filtrar-solicitacao

Hotfix/filtrar solicitacao

BrennoLyrio committed:
hotfix: Adicionando Skeleton

ajperes committed:
feat: Pagina para gerenciar Perfil

ajperes committed:
fix: Erro de undefined

ajperes committed:
feat: Filtro para Nome conforme spec.

ajperes committed:
feat: Filtro automatizado para Nome

ajperes committed:
refactor: Nome e Email com valores originais.

gui-benedito committed:
feat: decidir o tipo de solicitação de hora extra
se extra paga
se banco de horas
se recusado

gui-benedito committed:
feat: alterar lógica para renderizar modal

ajperes committed:
fix: Bug de campos e de forms dividios.

ajperes committed:
Bug: Nome e Email removidos depois de submit para Senha

gui-benedito committed:
Merge pull request #94 from SkyFlyTeam/feat/decidir-hora-extra

Feat/decidir hora extra

gui-benedito committed:
feat: solicitar hora extra

gui-benedito committed:
fix: tratar código do usuário

BrennoLyrio committed:
feat: Botao Dropdown Solicitacao Criado

BrennoLyrio committed:
hotfix: Botao Dropdown Solicitacao ajustado

gui-benedito committed:
fix: setas para mudar o mês

gustavomuraoka committed:
feat: adicionando modal de cadastro de férias

BrennoLyrio committed:
hotfix: problemas de conflitos

BrennoLyrio committed:
Merge pull request #90 from SkyFlyTeam/hotfix/skeleton

Hotfix/skeleton

gui-benedito committed:
fix: tratar componentes não utilizados no modal

gui-benedito committed:
Merge pull request #95 from SkyFlyTeam/feat/solicitar_hora_extra

Feat/solicitar hora extra

BrennoLyrio committed:
feat: Adicionando novo acesso a página de solicitações pelo Gestor

BrennoLyrio committed:
Hotfix: corrigindo para o merge

karengoncalves8 committed:
Feat/tela banco horas (#96)

* feat: adicionado o componente de datatable

* feat: adicionado modo horizontal a tabela e formatação de campos

* feat: adicionado setor/dados toggle e banco horas diário

* refactor: retirado useEffect de debug

* fix: ajustado exibição dos usuários na tabela

* fix: ajuste do import de bancoHoras interface

* fix: ajustado carga do usuário no banco diário

karengoncalves8 committed:
fix: ajuste no estilo e adicionado extra planejada no card de horario (#100)

gustavomuraoka committed:
Hotfix: arrumando erros funcionais relativos a empresa (#83)

Co-authored-by: karegoncalves8 <karengoncalves128@gmail.com>

ajperes committed:
Hotfix/tela cadastro errors (#49)

* refactor: Redireciona ao login após cadastro

* fix: Erros apontados para 'Tela de Cadastro'

* fix: Cargo agora obrigatorio

---------

Co-authored-by: karegoncalves8 <karengoncalves128@gmail.com>
Co-authored-by: Karen Gonçalves <105525498+karengoncalves8@users.noreply.github.com>

karengoncalves8 committed:
feat/banco-horas-card (#97)

* feat: adicionado componente card de banco de horas

* fix: adicionado condicional para renderização condicional do card de acordo com nivel acesso

* fix: ajuste na renderização do slider mediante a cancelamento

gustavomuraoka committed:
hotfix: exibição apenas de setores da empresa correspondente (#93)

Co-authored-by: Karen Gonçalves <105525498+karengoncalves8@users.noreply.github.com>
Co-authored-by: karegoncalves8 <karengoncalves128@gmail.com>

gustavomuraoka committed:
hotfix: alteração de tabela Meus Pontos para correta exibição da coluna observação

gustavomuraoka committed:
Merge branch 'develop' into hotfix/coluna-observacao

ajperes committed:
Merge branch 'feat/perfil' into develop

ajperes committed:
refactor: adaptado ao merge de 'feat/perfil'

ajperes committed:
Merge pull request #92 from SkyFlyTeam/feat/perfil

Feat/perfil

Inclui correções ao uso de dados no cookie.

BrennoLyrio committed:
hotfix: merge

BrennoLyrio committed:
Merge branch 'develop' of https://github.com/SkyFlyTeam/BeeOnTime-frontend into botao-solicitacao

BrennoLyrio committed:
hotfix: ajustes na cor e nos arquivos para adicionar o modal

BrennoLyrio committed:
mergeado

ericloumendes committed:
feat: Solicitação ausência

ericloumendes committed:
Merge branch 'develop' of https://github.com/SkyFlyTeam/BeeOnTime-frontend into feat/solicitacao_ausencia

ajperes committed:
feat: Editar colaborador

BrennoLyrio committed:
hotfix: ajuste do nome e caixa alta

ericloumendes committed:
fix: Ajustando modal

ericloumendes committed:
feat: solicitação ausência

Feat/solicitacao ausencia

BrennoLyrio committed:
Merge pull request #99 from SkyFlyTeam/feat/gestor-solicitacao

Feat: Solicitação do Gestor

gustavomuraoka committed:
feat: criação de modal de ferias para gestor aprovar ou recusar

BrennoLyrio committed:
hotfix: corrigido para o merge

BrennoLyrio committed:
Merge pull request #98 from SkyFlyTeam/botao-solicitacao

feat: Botao Dropdown Solicitacao Criado

gustavomuraoka committed:
Merge branch 'develop' into feat/modal-ferias-gestor

gustavomuraoka committed:
hotfix: alterações para melhor aderencia à develop

gustavomuraoka committed:
Merge pull request #106 from SkyFlyTeam/feat/modal-ferias-gestor

Criar modal de solicitação de férias | SKF-188 | RFN06

gustavomuraoka committed:
Merge branch 'develop' into hotfix/coluna-observacao

gustavomuraoka committed:
hotfix: arrumando para a visualização para mobile

gustavomuraoka committed:
Merge pull request #102 from SkyFlyTeam/hotfix/coluna-observacao

hotfix: alteração de tabela Meus Pontos para correta exibição da coluna Observação

BrennoLyrio committed:
refactor: retornando informações do commit e513e

gustavomuraoka committed:
hotfix: arrumando bug na aba de solicitações

gustavomuraoka committed:
Merge branch 'develop' into hotfix/bug-aba-solicitacao

gustavomuraoka committed:
Merge pull request #107 from SkyFlyTeam/hotfix/bug-aba-solicitacao

hotfix: arrumando bug na aba de solicitações

BrennoLyrio committed:
feat: Adicionando card de horas mensais trabalhadas (#101)

* feat: Adicionando card de horas mensais trabalhadas

* hotfix: Correções de conexão ao banco de horas

ericloumendes committed:
hotfix: Ajustando tabela para puxar dias sem ponto

ericloumendes committed:
Merge branch 'develop' of https://github.com/SkyFlyTeam/BeeOnTime-frontend into hotfix/tabela-pontos

ericloumendes committed:
hotfix: Removendo logs para testes

ericloumendes committed:
hotfix: Ajustando tabela de pontos para puxar datas sem pontos registrados

hotfix: Ajustando tabela de pontos para puxar datas sem pontos registrados

ericloumendes committed:
feat: Model licença médica

gui-benedito committed:
fix: tratar upload e download de arquivos

gui-benedito committed:
fix: mapear modais a serem renderizados

gui-benedito committed:
Merge pull request #109 from SkyFlyTeam/feat/Modal_licença_médica

feat: Model licença médica

gui-benedito committed:
renderizar modais

ericloumendes committed:
hotfix: Ajustando INVALID VALUE na tabela de pontos (#111)

karengoncalves8 committed:
Hotfix/solicitacao nivel acesso (#112)

* fix: ajustado gestor não aparecer dropdown de solicitação

* fix: ajustado tab

* fix: ajustado badge de pendente para o gestor

gui-benedito committed:
Merge pull request #113 from SkyFlyTeam/hotfix/modais_solicitacao

renderizar modais

gui-benedito committed:
hotfix: tratar modais decididos

gui-benedito committed:
Merge pull request #114 from SkyFlyTeam/hotfix/modais_solicitacao

hotfix: tratar modais decididos

ajperes committed:
refactor: Corrigido bugs e funcionalidade.

ajperes committed:
refactor: Nomes de interface importada para Usuario.

ajperes committed:
Merge pull request #104 from SkyFlyTeam/feat/editarColab

Feat/editar colab

karengoncalves8 committed:
fix: adicionado modelo de pr novamente (#117)

BrennoLyrio committed:
hotfix: deixando o nome conforme a empresa (#115)

* hotfix: deixando o nome conforme a empresa

* hotfix: empresa conforme o usuário

BrennoLyrio committed:
hotfix: tirando o nome no email (#165)

gui-benedito committed:
feat: tabela de falhas de marcação

gui-benedito committed:
feat: adicionar paginação na página de falhas

gui-benedito committed:
fix: corrigir url de requisição por setor

gui-benedito committed:
feat: alterar tabela na vertical para o responsivo

gui-benedito committed:
feat: alterar tabela na vertical para o responsivo

gui-benedito committed:
feat: renderizar gráfico

gui-benedito committed:
feat: gráficos de atrasos

gui-benedito committed:
fix: tratar a falta de dados

gui-benedito committed:
fix: tratar escolha de datas

andresalerno reviewed (dismissed):
Bom trabalho!

SarahBatagioti reviewed (changes requested):
- Quando a jornada é flexível, colocar Horário esperado o ultimo horário que ele poderia bater o ponto
![image](https://github.com/user-attachments/assets/30d2b518-e516-4bc9-a33c-868e63943c74)

- Aumentar o tamanho do gráfico no mobile 
![image](https://github.com/user-attachments/assets/ecd6286c-1925-41b2-824d-ac9d9c87467d)

- No sidebar deixar o texto em uma só linha (se diminuir 1px do tamanho já ajusta)
![image](https://github.com/user-attachments/assets/a8c8d489-2a03-4403-8a70-8b450a02a9ef)

gui-benedito committed:
fix: ajustar pontuações
'-' ao não tiver horário fixo
ajustar gráfico no mobile
ajustar tamanho do título no sidebar

gui-benedito committed:
fix: ajustar filtra de data e horario de batida esperado

SarahBatagioti reviewed (changes requested):
- A data de  "Até" vem vazia
![image](https://github.com/user-attachments/assets/5f3d6a50-645e-4e63-b46e-36077537293d)

gui-benedito committed:
fix: ajustar data fim
